kratos/includes/model_part.h

```
@@ -469,8 +469,11 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     template<class TDataType>
     void AddNodalSolutionStepVariable(Variable<TDataType> const& ThisVariable)
     {
-        KRATOS_ERROR_IF((this->GetRootModelPart()).Nodes().size() != 0) << "attempting to add the variable: " << ThisVariable.Name() << " to the model part with name: " << this->Name() << " which is not empty" << std::endl;
-        mpVariablesList->Add(ThisVariable);
+        if (!mpVariablesList->Has(ThisVariable))
```

oh i see.

however i am not sure if we should throw an error at all here. maybe we should create the mpVariableList if it does not exist. @KratosMultiphysics/technical-committee opinions?

---

@RiccardoRossi  we had quite some problems before this error was added (done by you btw), see #492

---

jeje, i have short memory...thx for the pointer